lib/lapin/connection.ex

```diff
@@ -443,6 +443,16 @@ defmodule Lapin.Connection do
         {:error, reason} ->
           Logger.debug("Failed rejecting message #{delivery_tag}: #{inspect(reason)}")
       end
+  catch
+    :exit, reason ->
+      case Consumer.reject_message(consumer, delivery_tag, not redelivered) do
+        :ok ->
+          Logger.error("Rejected message #{delivery_tag} due to process exit: #{inspect(reason)}")
+          :ok
+
+        {:error, reject_error} ->
+          Logger.debug("Failed rejecting message #{delivery_tag}: #{inspect(reject_error)}")
```

Comment:
```suggestion
          Logger.error("Failed rejecting message #{delivery_tag}: #{inspect(reject_error)}")
```

And we should def have it in Sentry 🤔 Maybe we should let it burn in this case?

Reply:
If we were to let it burn then [this `spawn` here](https://github.com/surferseo/lapin/blob/stability-improvements/lib/lapin/connection.ex#L336) would bite us again